portal/import_template.php

```diff
@@ -12,106 +12,227 @@
+        }
+        renderEditorHtml($_REQUEST['docid'], $template_content);
+    } else {
+        die(xlt('Invalid File'));
     }
-
-    return $rebuiltPath;
+} elseif (!empty($_GET['templateHtml'] ?? null)) {
+    renderEditorHtml($_REQUEST['docid'], $_GET['templateHtml']);
 }
 
-function redirect()
+function renderEditorHtml($template_id, $content)
 {
-    header('HTTP/1.0 404 Not Found');
-    die();
-}
+    $lists = [
+        '{ParseAsHTML}', '{TextInput}', '{sizedTextInput:120px}', '{smTextInput}', '{TextBox:03x080}', '{CheckMark}', '{ynRadioGroup}', '{TrueFalseRadioGroup}', '{DatePicker}', '{DateTimePicker}', '{StandardDatePicker}', '{CurrentDate:"global"}', '{CurrentTime}', '{DOS}', '{ReferringDOC}', '{PatientID}', '{PatientName}', '{PatientSex}', '{PatientDOB}', '{PatientPhone}', '{Address}', '{City}', '{State}', '{Zip}', '{PatientSignature}', '{AdminSignature}', '{AcknowledgePdf: : }', '{EncounterForm:LBF}', '{Medications}', '{ProblemList}', '{Allergies}', '{ChiefComplaint}', '{DEM: }', '{HIS: }', '{LBF: }', '{GRP}{/GRP}'
+    ];
+    ?>
+    <!DOCTYPE html>
+    <html>
+    <head>
+        <?php Header::setupHeader(['ckeditor']); ?>
+    </head>
+    <style>
+      input:focus,
+      input:active
+      {
+        outline:0px !important;
+        -webkit-appearance:none;
+        box-shadow: none !important;
+      }
+      .list-group-item {
+        font-size: .9rem;
+      }
+    </style>
+    <body>
+        <div class="container-fluid mt-2">
+            <div class="row">
+                <div class="col-10 px-1">
+                    <form class="sticky-top" action='./import_template.php' method='post'>
+                        <input type="hidden" name="docid" value="<?php echo $template_id ?>">
+                        <input type='hidden' name='mode' value="save">
+                        <input type='hidden' name='service' value='window'>
+                        <textarea class='inline-editor' contenteditable='true' cols='80' rows='10' id='templateContent' name='content'><?php echo $content ?></textarea>
+                        <div class="row btn-group mt-1 float-right">
+                            <div class='col btn-group mt-1 float-right'>
+                                <button type="submit" class="btn btn-sm btn-primary"><?php echo xlt("Save"); ?></button>
+                                <button type='button' class='btn btn-sm btn-secondary' onclick='parent.window.close() || parent.dlgclose()'><?php echo xlt('Cancel'); ?></button>
+                            </div>
+                        </div>
+                    </form>
+                </div>
+                <div class="col-sm-2 px-0">
+                    <div class='h4'><?php echo xlt("Directives") ?></div>
+                    <ul class='list-group list-group-flush pl-1 mb-5'>
+                        <?php
+                        foreach ($lists as $list) {
+                            echo "<input class='list-group-item p-1' value='$list'>";
```

Comment:
attr() around $list

Reply:
My mom said I didn't have to escape any longer